src/pcm_lowlevel_u585ai.cpp

```diff
@@ -0,0 +1,407 @@
+    int32_t Z = ((DMic_OnBoard_Buffer[j + ((PCM.SampleRate / 1000U) * N_MS_PER_INTERRUPT)]) * (int32_t)(PCM.Volume)) / 100;
+    PCM.Buffer[j] = (uint16_t) SaturaLH(Z, -32760, 32760);
+  }
+
+  if (PCM.OnRecv != nullptr) {
+    PCM.OnRecv();
+  }
+}
+
+/**
+* @brief  Half regular conversion complete callback.
+* @param  hmdf   MDF filter handle.
+* @retval None
+*/
+void HAL_MDF_AcqHalfCpltCallback(MDF_HandleTypeDef *hmdf)
+{
+  UNUSED(hmdf);
+
+  if (PCM.Buffer == nullptr) {
+    return;
+  }
+
+  if (DMic_OnBoard_CountSkip < 64) {
+    ++DMic_OnBoard_CountSkip;
+    return;
+  }
+
+  for (uint32_t j = 0U; j < ((PCM.SampleRate / 1000U) * N_MS_PER_INTERRUPT); j++) {
+    int32_t Z = (DMic_OnBoard_Buffer[j] * (int32_t)(PCM.Volume)) / 100;
+    PCM.Buffer[j] = (uint16_t) SaturaLH(Z, -32760, 32760);
+  }
+
+  if (PCM.OnRecv != nullptr) {
+    PCM.OnRecv();
+  }
+}
+
+/**
+  * @brief System Clock Configuration
+  * @retval None
+  */
+void SystemClock_Config(void)
+{
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+
+  /** Configure the main internal regulator output voltage */
+  if (HAL_PWREx_ControlVoltageScaling(PWR_REGULATOR_VOLTAGE_SCALE1) != HAL_OK) {
+    Error_Handler();
+  }
+
+  /** Initializes the CPU, AHB and APB buses clocks */
+  RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_MSI | RCC_OSCILLATORTYPE_HSI48 | RCC_OSCILLATORTYPE_HSE;
+
+  // RCC_OscInitStruct.HSIState = RCC_HSI_ON;
+  RCC_OscInitStruct.HSEState = RCC_HSE_ON;
+  RCC_OscInitStruct.HSI48State = RCC_HSI48_ON;
+  RCC_OscInitStruct.HSICalibrationValue = RCC_HSICALIBRATION_DEFAULT;
+  RCC_OscInitStruct.MSIState = RCC_MSI_ON;
+  RCC_OscInitStruct.MSICalibrationValue = RCC_MSICALIBRATION_DEFAULT;
+  RCC_OscInitStruct.MSIClockRange = RCC_MSIRANGE_4;
+  RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
+  RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_MSI;
+  RCC_OscInitStruct.PLL.PLLMBOOST = RCC_PLLMBOOST_DIV1;
+
+  RCC_OscInitStruct.PLL.PLLM = 1;
+  RCC_OscInitStruct.PLL.PLLN = 80;
+  RCC_OscInitStruct.PLL.PLLP = 8;
+  RCC_OscInitStruct.PLL.PLLQ = 2;
+  RCC_OscInitStruct.PLL.PLLR = 2;
+
+  RCC_OscInitStruct.PLL.PLLRGE = RCC_PLLVCIRANGE_0;
+  RCC_OscInitStruct.PLL.PLLFRACN = 0;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
+    Error_Handler();
+  }
+
+  /** Initializes the CPU, AHB and APB buses clocks */
+  RCC_ClkInitStruct.ClockType =
+    RCC_CLOCKTYPE_HCLK | RCC_CLOCKTYPE_SYSCLK
+    | RCC_CLOCKTYPE_PCLK1 | RCC_CLOCKTYPE_PCLK2
+    | RCC_CLOCKTYPE_PCLK3;
+  RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
+  RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
+  RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV1;
+  RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV1;
+  RCC_ClkInitStruct.APB3CLKDivider = RCC_HCLK_DIV1;
+
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_4) != HAL_OK) {
+    Error_Handler();
+  }
+  __HAL_RCC_PWR_CLK_DISABLE();
```

Comment:
I would not recommend to disable it as you don't know if someone else enable it before.

Reply:
This code was not created by me, I have absolutely no idea of what it does.

Reply:
I think that we can remove line 277 without any issue

Reply:
As example, other libary enable the clk of the PWR domain and assume it is on while this library simply disable it.